### PR TITLE
Remove tool wrapper

### PR DESCRIPTION
**Pull request by solsjo**

### CODE REVIEW

**Comment by solsjo** [comment]
@Kernald @patbro, when you have time. Or someone else with merge permissions. This chain of commits greatly simplifies rule logic and at the same time reduces build times of build targets.

**Comment by patbro** [comment]
Thanks for the trigger! I am still AFK, will probably get to this tomorrow.

**Comment by solsjo** [comment]
@patbro Thanks! :) No stress!

**Comment by Kernald** [comment]
I take it the first commit here is the one from #109?

**Comment by solsjo** [comment]
Yes!

**Comment by patbro** [comment]
> @Kernald @patbro, when you have time. Or someone else with merge permissions. This chain of commits greatly simplifies rule logic and at the same time reduces build times of build targets.

I believe you know better what you're doing than me ;-) No comments whatsoever from my side, I will leave it up to @Kernald  to merge it.

**Comment by solsjo** [comment]
Don't say that. My path to hubris is known to be short. :D Thanks for the review @patbro

File: example/BUILD.bazel
Comment (solsjo) [inline example/BUILD.bazel]: Remove? There is both bibtex and biber, so we don't want to hardcode the choice of binary internally.

But latexrun, runs the bib task with cwd set to the output directory, which puts a burden on the user to point to it correctly. Perhaps it's better to add a default parameter to the rule api? If there is only two possible options?

On the other hand, if the only option we currently had is biber. Then, I think this option is fine. And the line should be removed.

File: latex.bzl
Comment (solsjo) [inline latex.bzl]: Repetitions, make a function to encapsulate?

File: latex.bzl
Comment (solsjo) [inline latex.bzl]: dep_tools doesn't make much sense anymore when the tool wrapper is gone, consider dropping it and use the toolchain directly

File: latex.bzl
Comment (solsjo) [inline latex.bzl]: Can we use "." Instead of PWD?

File: latex.bzl
Comment (solsjo) [inline latex.bzl:135]: Here we have both bib tools so it makes sense to update the api to support both with a dedicated parameter?

**Review by solsjo — COMMENTED** [review]
Removing setting the bibtex binary is a breaking change. We should at least provide an option to switch between the existing bib binaries

File: latex.bzl
Comment (solsjo) [reply latex.bzl:135]: done

File: latex.bzl
Comment (solsjo) [reply latex.bzl]: done

File: latex.bzl
Comment (solsjo) [reply latex.bzl]: done

File: latex.bzl
Comment (solsjo) [reply latex.bzl]: done

File: example/BUILD.bazel
Comment (solsjo) [reply example/BUILD.bazel]: done

**Comment by solsjo** [comment]
@Kernald , when you have time :)

File: example/BUILD.bazel
Comment (Kernald) [inline example/BUILD.bazel]: Nit: `default`

File: latex.bzl
Comment (Kernald) [inline latex.bzl]: Nit: `invocation`

File: latex.bzl
Comment (Kernald) [inline latex.bzl:108]: What is Python needed for anymore? If we still need it, can we provide it through rules-python instead?

File: latex_engine_cmds.bzl
Comment (Kernald) [inline latex_engine_cmds.bzl]: You could use [config_setting](https://bazel.build/reference/be/general#config_setting) with `compilation_mode` for this

File: latex_engine_cmds.bzl
Comment (Kernald) [inline latex_engine_cmds.bzl]: This should probably be cleaned up

File: latex_engine_cmds.bzl
Comment (Kernald) [inline latex_engine_cmds.bzl]: The same `config_setting` could be used here

File: latex.bzl
Comment (solsjo) [reply latex.bzl:108]: It should be possible to remove.
IIRC, this was only due latexrun being provided through exports_files.

I can change it to py_binary instead

Comment (solsjo) [reply latex.bzl:108]: changed to py_binary, yet, it fails, I wonder if setting the env parameter overwrites some default env values...

File: latex_engine_cmds.bzl
Comment (solsjo) [reply latex_engine_cmds.bzl]: Let me think about it, the debug logging output doesn't really affect the artifact output. So it doesn't really behave as e.g. opt or dbg for cc-rules in that sense, right?

File: latex_engine_cmds.bzl
Comment (solsjo) [reply latex_engine_cmds.bzl]: will do

File: latex.bzl
Comment (solsjo) [reply latex.bzl:108]: This is related to:
https://github.com/bazelbuild/bazel/issues/4912
ctx.configuration.default_shell_env used to contain the default env such that it could be read and modified appended to. But is currently not.

I still think I prefer this over a wrapper. This should be ironed out any year now :P

File: latex_engine_cmds.bzl
Comment (solsjo) [reply latex_engine_cmds.bzl]: removing the debug TODOs for now, I can write an issue about it later and we can discuss pros- and cons there.

File: latex_engine_cmds.bzl
Comment (solsjo) [reply latex_engine_cmds.bzl]: answered in the last comment

File: latex.bzl
Comment (solsjo) [reply latex.bzl:108]: I guess, as you mention, if we define (maybe build) our own interpreter, that we won't need to depend on the host python interpreter.

But maybe that is for another pull-request

File: latex_engine_cmds.bzl
Comment (solsjo) [reply latex_engine_cmds.bzl]: Yeah, maybe you are right. 😊 But I'll add it for another pull request

**Comment by solsjo** [comment]
@Kernald, when you have time :)

File: latex.bzl
Comment (Kernald) [reply latex.bzl:108]: Sounds good for now.

**Review by Kernald — APPROVED** [review]

**Pull request merged**
